src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -534,14 +528,24 @@ public void StopStake()
                         utxoStakeDescription.Secret = walletSecret; // Temporary.
                         utxoStakeDescriptions.Add(utxoStakeDescription);
                         totalBalance += utxo.Value;
+
                         this.logger.LogTrace("UTXO '{0}' with value {1} might be available for staking.", utxoStakeDescription.OutPoint, utxo.Value);
                     }
                 }
 
                 this.logger.LogTrace("Wallet contains {0} coins.", new Money(totalBalance));
 
                 if (blockTemplate == null)
-                    blockTemplate = this.blockAssemblerFactory.Create(chainTip, new AssemblerOptions() { IsProofOfStake = true }).CreateNewBlock(new Script());
+                    blockTemplate = new PosBlockAssembler(
```

Comment:
I think it will be a good idea to create a single PosBlockAssembler somewhere in constructor of PoSMinting or when staking starts and reuse it instead of recreating PosBlockAssembler every time consensus advances.

Reply:
I agree, the previous version of this code created a assembler every time which was wrong. This upgrade remove the factory, but did not address the main problem. Fixing the chain tip is only needed when we create the new block. Therefore it should not go into the constructor.

Reply:
Yes, sure 🙂